crates/starknet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -232,26 +241,21 @@ impl JsonRpcHandler {
         block_id: BlockId,
         contract_address: ContractAddress,
     ) -> RpcResult<ClassHash> {
-        let starknet = self.api.starknet.read().await;
-        match starknet.get_class_hash_at(&block_id.into(), &contract_address) {
+        match self.api.starknet.write().await.get_class_hash_at(block_id.into(), contract_address) {
```

Comment:
Hear and in other places (`get_class`, `get_class_at`), why lock this with `write` privilege instead of read?

Reply:
Old implementation tapped in CachedState which used &mut state for `get_class_hash_at`. Now reverted to read access